### ISSUE TITLE
Difficulty building from "Meet Swift Godot" as well as from fresh `SwiftGodotKick` project

### ISSUE THREAD

**Issue by JacobSyndeo** [migueldeicaza/SwiftGodot#514]
**Describe the bug**
I'm having some trouble compiling when following the ["Meet Swift Godot" tutorial](https://migueldeicaza.github.io/SwiftGodotDocs/tutorials/swiftgodot/your-first-extension), as well as when running `make` on a fresh [SwiftGodotKick](https://github.com/EstevanBR/SwiftGodotKick) project. The same error appears in both cases, specifically:

`external macro implementation type 'SwiftGodotMacroLibrary.GodotMacro' could not be found for macro 'Godot'`

Suspicious areas of the logs:
- `Generator/default_value: no support for [Variant] = 0` 4 times, followed by `Generator/default_value: no support for [typedarray::RDPipelineSpecializationConstant] = Array[RDPipelineSpecializationConstant]([])
`. I'm thinking that the other issues may just be symptoms of whatever is causing this particular error.
- Compiler complaints about Rosetta: `rosetta error: Failed to open libRosettaRuntime`. However, according to Activity Monitor, I do have an active process running via Rosetta (the Fish shell), so it's not that I don't have Rosetta installed. Something else seems to be wrong.

Occam's razor has me supposing that the issue is somewhere on my end, but I'm hoping that someone more knowledgeable with `SwiftGodot` may be able to point me in the right direction here.

**To Reproduce**
Follow the ["Meet Swift Godot" tutorial](https://migueldeicaza.github.io/SwiftGodotDocs/tutorials/swiftgodot/your-first-extension) up to Step 4

**Expected behavior**
Successful compilation

**Screenshots**

**Desktop (please complete the following information):**
 - macOS Sonoma 14.2.1
 - M1 chip
 - Swift 5.10

**Additional context**
Build log:
```
swift build
Building for debugging...
[1/1] Write swift-version--58304C5D6DBC2206.txt
Build complete! (0.83s)
Building for debugging...
Running with projectDir=$(projectDir) and output=/Users/jacobpritchett/Developer/SimpleRunnerDriver/.build/plugins/outputs/swiftgodot/SwiftGodot/CodeGeneratorPlugin/GeneratedSources
Generator/default_value: no support for [Variant] = 0
Generator/default_value: no support for [Variant] = 0
Generator/default_value: no support for [Variant] = 0
Generator/default_value: no support for [Variant] = 0
Generator/default_value: no support for [typedarray::RDPipelineSpecializationConstant] = Array[RDPipelineSpecializationConstant]([])
Generator/default_value: no support for [typedarray::RDPipelineSpecializationConstant] = Array[RDPipelineSpecializationConstant]([])
error: emit-module command failed with exit code 1 (use -v to see invocation)
rosetta error: Failed to open libRosettaRuntime
 /Users/jacobpritchett/Developer/SimpleRunnerDriver/Sources/SimpleRunnerDriver/PlayerController.swift:12:7: error: external macro implementation type 'SwiftGodotMacroLibrary.GodotMacro' could not be found for macro 'Godot'
class PlayerController: CharacterBody2D {
      ^
/Users/jacobpritchett/Developer/SimpleRunnerDriver/.build/checkouts/SwiftGodot/Sources/SwiftGodot/MacroDefs.swift:22:14: note: 'Godot' declared here
public macro Godot(_ behavior: ClassBehavior = .gameplay) = #externalMacro(module: "SwiftGodotMacroLibrary", type: "GodotMacro")
             ^
/Users/jacobpritchett/Developer/SimpleRunnerDriver/Sources/SimpleRunnerDriver/SimpleRunnerDriver.swift:7:1: error: external macro implementation type 'SwiftGodotMacroLibrary.InitSwiftExtensionMacro' could not be found for macro 'initSwiftExtension(cdecl:types:)'
#initSwiftExtension(cdecl: "swift_entry_point", types: [PlayerController.self])
^
/Users/jacobpritchett/Developer/SimpleRunnerDriver/.build/checkouts/SwiftGodot/Sources/SwiftGodot/MacroDefs.swift:118:14: note: 'initSwiftExtension(cdecl:types:)' declared here
public macro initSwiftExtension(cdecl: String,
             ^
rosetta error: Failed to open libRosettaRuntime
 /Users/jacobpritchett/Developer/SimpleRunnerDriver/Sources/SimpleRunnerDriver/PlayerController.swift:12:7: error: external macro implementation type 'SwiftGodotMacroLibrary.GodotMacro' could not be found for macro 'Godot'
class PlayerController: CharacterBody2D {
      ^
/Users/jacobpritchett/Developer/SimpleRunnerDriver/.build/checkouts/SwiftGodot/Sources/SwiftGodot/MacroDefs.swift:22:14: note: 'Godot' declared here
public macro Godot(_ behavior: ClassBehavior = .gameplay) = #externalMacro(module: "SwiftGodotMacroLibrary", type: "GodotMacro")
             ^
rosetta error: Failed to open libRosettaRuntime
 /Users/jacobpritchett/Developer/SimpleRunnerDriver/Sources/SimpleRunnerDriver/SimpleRunnerDriver.swift:7:1: error: external macro implementation type 'SwiftGodotMacroLibrary.InitSwiftExtensionMacro' could not be found for macro 'initSwiftExtension(cdecl:types:)'
#initSwiftExtension(cdecl: "swift_entry_point", types: [PlayerController.self])
^
/Users/jacobpritchett/Developer/SimpleRunnerDriver/.build/checkouts/SwiftGodot/Sources/SwiftGodot/MacroDefs.swift:118:14: note: 'initSwiftExtension(cdecl:types:)' declared here
public macro initSwiftExtension(cdecl: String,
             ^
error: fatalError
```

Any help would be greatly appreciated!

**Comment by migueldeicaza** [comment]
The "Generator/default_value" are just a message for me, it is a reminder that I have to support a handful of additional things.

What is surprising is that you are triggering Rosetta, as this should be a fully native build, I am confused as to how this could end up this way.   There should be no need to run Rosetta at all, which would indicate that you have some Intel code being emulated.

**Comment by JacobSyndeo** [comment]
Went down a bit of a rabbit hole here… as it turns out, that bit in my first message about the Fish shell running via Rosetta was a sign.

Turns out that my Homebrew installation was still x86_64, as I bought and set up this Mac right as Apple Silicon was launched—before Homebrew was ported over for arm64. Running `arch` returned "i386"… clearly not what it's supposed to say. I'll need to [migrate my Homebrew installation](https://blog.smittytone.net/2021/02/07/how-to-migrate-to-native-homebrew-on-an-m1-mac/) now.

In the meantime, I tried on a newer Mac and had no issues, SwiftGodot runs great! 👍🏼

So hopefully this can help anyone else who arrives here via a web search for that Rosetta error: Make sure you're not accidentally using Intel binaries anywhere in the stack—double-check by running `arch`!

And thanks @migueldeicaza (and other contributors) for building and maintaining this project! You're doing the Swift and Godot communities a great service here; I'm really excited to build with it.
